### PR TITLE
Added correction to KullbackLeibler

### PR DESCRIPTION
**Pull request by Alexander-Sol**
Kullback leibler now increases score in case of non-matching peaks

### CODE REVIEW

File: mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs [hunk 1]
```diff
@@ -173,6 +173,67 @@ private List<(double, double)> IntensityPairs(bool allPeaks)
             return intensityPairs;
         }
 
+        //This function is used for calculating correctedIntensityPairs after renormalization
+        private List<(double, double)> CorrectedIntensityPairs(double[] tempExperimentalY, double[] tempTheoreticalY, bool allPeaks)
```
Comment (trishorts) [inline mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs:177]: can we just call the original intensity pairs function w/ the revised arrays? i can't tell immediately if this is just duplicated code.

File: mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs [hunk 2]
```diff
@@ -344,18 +405,48 @@ private double[] NormalizeSpectrumSum(double[] spectrum)
...
+                    }
+                }
             }
+            else
+            {
+                // Add correctionConstant and renormalize
+                // Need to use temp variables to avoid modifiying the Y array fields
+                double[] tempExperimentalYArray = Normalize(ExperimentalYArray.Select(i => i + correctionConstant).ToArray(), SpectrumNormalizationScheme.spectrumSum);
+                double[] tempTheoreticalYArray = Normalize(TheoreticalYArray.Select(i => i + correctionConstant).ToArray(), SpectrumNormalizationScheme.spectrumSum);
+                List<(double, double)> correctedIntensityPairs = CorrectedIntensityPairs(tempExperimentalYArray, tempTheoreticalYArray, allPeaks: true);
```
Comment (trishorts) [inline mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs:439]: can we just call the original intensity pairs function?

Comment (Alexander-Sol) [reply mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs]: Original intensity pairs function wouldn't take any arguments, instead pulling from the object's fields. I've edited the original function to have optional arguments.

File: mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs (hunk 1 above)
Comment (Alexander-Sol) [reply mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs]: Deleted this function

**Review by avcarr2 — DISMISSED** [review]
Looks good.

**Review by nbollis — APPROVED** [review]

**Review by nbollis — APPROVED** [review]

**Review by weaversd — APPROVED** [review]

**Pull request merged**